backend/http/http.go

```diff
@@ -601,13 +591,22 @@ func (o *Object) stat(ctx context.Context) error {
 	if err != nil {
 		return fmt.Errorf("failed to stat: %w", err)
 	}
+	return o.stat(ctx, res)
+}
+
+// stat updates info fields in the Object according to HTTP response headers
+func (o *Object) stat(ctx context.Context, res *http.Response) error {
```

Comment:
In other backends we call this function `decodeMetadata` which might be a better description of what it does, or maybe `decodeResponse`. I don't like `stat` because that implies to me that it is fetching the info like the `os.Stat` call.